modules/standard/IO.chpl

```diff
@@ -5671,6 +5824,411 @@ proc _channel.readLine(type t=string, maxSize=-1, stripNewline=false): t throws
   return retval;
 }
 
+
+/*
+  Read until the given separator is found, returning the contents of the
+  ``fileReader`` through that point.
+
+  If the separator is found in the next ``maxSize`` codepoints/bytes, the input
+  marker is left immediately after it. If it isn't found, a ``BadFormatError``
+  is thrown and the input marker is left in its original position.
```

Comment:
This makes sense for the `t` returning overload, but I'm not 100% sure about throwing UnexpectedEOF from the `bool` returning overloads, as it would force this pattern:
```chapel
iter numbers() {
  var s: string;
  while r.readThrough(",", s) {
    yield s: int;
  }
}
```
to look like:
```chapel
iter numbers() {
  var s: string {
  try {
    while r.readThrough(",", s) {
      yield s:int;
    }
  } catch {
    return;
  }
}
```
Does that seem correct to you?

Reply:
I was trying to distinguish between the two things that could be thrown. Now that you mention it, I think it shouldn't throw if the separator isn't found and EOF is reached. I think we can make it consistent with `readLine` in that way -- which, IIRC, will just return the data without a terminating newline in the event of EOF.

Reply:
(And, I think the `t` returning version and the `ref arg: string` version should behave the same in this regard, whatever we pick).

Reply:
Yes, that's right.